### PR TITLE
server: add split support.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut  @disksing

### CODE REVIEW

File: server/cluster_worker_test.go [hunk 1]
```diff
@@ -442,3 +497,52 @@ func (s *testClusterWorkerSuite) TestBaseChangePeer(c *C) {
...
+	c.Assert(left.GetStartKey(), BytesEquals, []byte(""))
+	c.Assert(left.GetEndKey(), BytesEquals, []byte("b"))
+	c.Assert(left.GetRegionId(), Equals, region.GetRegionId())
+
+	right, err := cluster.GetRegion([]byte("b"))
+	c.Assert(err, IsNil)
+	c.Assert(right.GetStartKey(), BytesEquals, []byte("b"))
+	c.Assert(right.GetEndKey(), BytesEquals, []byte(""))
+
+	region, err = cluster.GetRegion([]byte("b"))
```
Comment (disksing) [inline server/cluster_worker_test.go]: Why testing "b" twice? Maybe you want to test "c"?

Comment (siddontang) [inline server/cluster_worker_test.go]: my fault.

**Comment by disksing** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @ngaut

File: server/cluster_worker_test.go [hunk 2]
```diff
@@ -307,8 +307,62 @@ func (s *mockRaftStore) handleChangePeer(c *C, req *raft_cmdpb.RaftCommandReques
...
+		}
+
+		peer.PeerId = proto.Uint64(id)
+		if id > maxPeerID {
+			maxPeerID = id
+		}
+		newRegion.Peers[i] = &peer
+	}
+
+	newRegion.MaxPeerId = proto.Uint64(maxPeerID)
```
Comment (ngaut) [inline server/cluster_worker_test.go]: We don't need to send max peer id. The raft group can calculate it automatically.

Comment (siddontang) [inline server/cluster_worker_test.go]: Here simulates the raft group, so it should handle here.

File: server/cluster_worker_test.go [hunk 3]
```diff
@@ -442,3 +497,52 @@ func (s *testClusterWorkerSuite) TestBaseChangePeer(c *C) {
 
 	s.checkRegionPeerNumber(c, regionKey, 3)
 }
+
+func (s *testClusterWorkerSuite) TestBaseSplit(c *C) {
```
Comment (ngaut) [inline server/cluster_worker_test.go]: s/TestBaseSplit/TestSplit

**Comment by ngaut** [comment]
LGTM

**Pull request merged**